tests/e2e/sync_test.go

```diff
@@ -0,0 +1,114 @@
+//go:build e2e_tests
+// +build e2e_tests
```

Comment:
You don't need both, the first line should suffice (with Go 1.17).
nit: let's call this "integration"